### PR TITLE
MNIST examples

### PR DESCRIPTION
**Pull request by ArneNx**
Refactoring and rewrite of the toy-examples to perform a simple MNIST training.

### CODE REVIEW

File: nnfabrik/templates/transfer/__init__.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import os
 from nnfabrik.main import Model, Dataset, Trainer, Seed, Fabrikant
 from nnfabrik.utility.dj_helpers import gitlog, make_hash
-from .trained_model import TrainedModelBase
+from nnfabrik.templates.trained_model import TrainedModelBase
 
 
 class TransferredTrainedModelBase(TrainedModelBase):
```
Comment (ArneNx) [inline nnfabrik/templates/transfer/__init__.py:10]: This should be part of another PR

**Review by ArneNx — COMMENTED** [review]
`transfer/__init__.py` shouldn't be part of this PR

File: nnfabrik/examples/mnist/trainer.py [hunk 2]
```diff
@@ -13,31 +14,42 @@ def __init__(self, model, dataloaders, seed, epochs=5):
...
+                x_flat = x.flatten(1, -1)  # treat the images as flat vectors
+                y_hat = self.model(x_flat)
                 loss = self.loss_fn(y_hat, y)
+                # backward:
                 loss.backward()
                 self.optimizer.step()
-                _losses.append(loss.item())
-
-        losses.append(np.mean(_losses))
+                # keep track of accuracy:
```
Comment (cblessing24) [inline nnfabrik/examples/mnist/trainer.py:37]: I think the code is self-explanatory and the comment is not really needed.

Comment (ArneNx) [reply nnfabrik/examples/mnist/trainer.py:37]: This is meant more to structure the code into forward, backward and tracking.

Comment (cblessing24) [reply nnfabrik/examples/mnist/trainer.py:37]: I see. In that case I would personally prefer having separate (private) functions for forward backward and tracking instead of commented sections. That said I am also fine with it staying as it is.

Comment (ArneNx) [reply nnfabrik/examples/mnist/trainer.py:37]: That's again a trade-off between good practices and compact code. I opted for the latter here.

**Review by mohammadbashiri — APPROVED** [review]
